### PR TITLE
Make keepalived role more configurable (issue #683)

### PR DESCRIPTION
**Pull request by abyss-ms**
Issue reference: #683 

This change shouldn't affect the working environment, so, by default, the keepalived configuration remains the same. These fixes make the keepalived role configurable. They also update the configuration to comply with modern Ansible linters.

### CODE REVIEW

File: roles/keepalived/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    interface: "{{ vip_interface }}"
+    virtual_router_id: "{{ keepalived_virtual_router_id | default(123) }}"
+    priority: 100
+    advert_int: 2
+    check_status_command: /usr/libexec/keepalived/haproxy_check.sh
+    authentication:
+      auth_type: PASS
+      auth_pass: "1ce24b6e"
+    virtual_ipaddresses:
+      - "{{ cluster_vip  }}"
```
Comment (vitabaks) [inline roles/keepalived/defaults/main.yml:15]:
```suggestion
      - "{{ cluster_vip }}"
```

File: roles/keepalived/tasks/main.yml [hunk 2]
```diff
@@ -42,15 +43,29 @@
...
   notify: "restart keepalived"
+  vars:
+    keepalived_instances: "{{ keepalived_instances_parsed }}"
   when: add_balancer is not defined or not add_balancer|bool
   tags: keepalived_conf, keepalived
 
-- block:  # for add_balancer.yml
+- name: Generate conf file "/etc/keepalived/keepalived.conf" for add_balancer
+  when: add_balancer is defined and add_balancer|bool
+  tags: keepalived_conf, keepalived
```
Comment (vitabaks) [inline roles/keepalived/tasks/main.yml:67]: please stick to one code format. Move the conditions and tags down in the block.

Comment (abyss-ms) [reply roles/keepalived/tasks/main.yml]: Done :)

File: roles/keepalived/tasks/main.yml [hunk 3]
```diff
@@ -42,11 +43,22 @@
   notify: "restart keepalived"
   tags: keepalived_conf, keepalived
 
+- name: Parse keepalived_instances as YAML
+  ansible.builtin.set_fact:
+    keepalived_instances_parsed: "{{ keepalived_instances | from_yaml }}"
```
Comment (vitabaks) [inline roles/keepalived/tasks/main.yml:48]: this task seems superfluous, why are you using the `from_yaml` filter here?

Comment (abyss-ms) [reply roles/keepalived/tasks/main.yml:48]: I'm using a Jinja2 template to create the keepalived_instances variable in my group_vars. Indeed, this filter should be done in group_vars, not here.

**Pull request merged**